server.js

```diff
@@ -23,6 +40,54 @@ app.get('/', (req, res) => {
   res.send('Hello world')
 })
 
+// THOUGHTS
+app.get('/thoughts', async (req, res) => {
+
+  try {
+    //Success
+    const thoughts = await Thought.find().sort({ createdAt: 'desc', heart: +1 }).limit(20).exec()
+    res.status(200).json(thoughts)
+  } catch (err) {
+    res.status(400).json({ message: 'Could not get thoughts', error: err.errors })
+  }
+})
+
+// POST A THOUGHT
+app.post('/thoughts', async (req, res) => {
+  //Retrieve the information sent by the client to our API endpoint
+  const { message } = req.body
+  //use our mongoose model to create the database entry
+  const thought = new Thought({ message })
+
+  try {
+    //Success
+    const savedThought = await thought.save()
+    res.status(201).json(savedThought)
+  } catch (err) {
+    res.status(400).json({ message: 'Could not save your thought to the Database', error: err.errors })
+  }
+})
+
+// LIKE A THOUGHT
+app.post('/:thoughtId/like', async (req, res) => {
+  //so this endpoint should update the heart Number
+
+  try {
+    //Success
+    const like = await Thought.findOneAndUpdate(
+      { "_id": req.params.thoughtId },//filter
+      { $inc: { "heart": 1 } },//update
+      { returnNewDocument: true }//doesn't update/work
+
+    )
+    res.status(201).json(like)
```

Comment:
I like that you returned the new thought here!  Hopefully this worked as expected!
This will allow the frontend to rely on the backend for the updated thought information